crates/red_knot_python_semantic/src/types.rs

```diff
@@ -310,7 +310,7 @@ impl<'db> Type<'db> {
         matches!(self, Type::Never)
     }
 
-    fn is_none(&self, db: &'db dyn Db) -> bool {
+    pub fn is_none(&self, db: &'db dyn Db) -> bool {
```

Comment:
I don't think this change is necessary with the latest version of this PR

```suggestion
    fn is_none(&self, db: &'db dyn Db) -> bool {
```